### PR TITLE
fix(): Don't expose EventEmitters in Component public API

### PR DESCRIPTION
**Pull request by trik**
refers #455

### CODE REVIEW

**Comment by trik** [comment]
@DevVersion @jelbourn ok, submitted the clean one, sorry about that

File: src/components/checkbox/checkbox.ts [hunk 1]
```diff
@@ -273,9 +275,9 @@ export class MdCheckbox implements AfterContentInit, ControlValueAccessor {
     // We always have to stop propagation on the change event.
     // Otherwise the change event, from the input element, will bubble up and
     // emit its event object to the `change` output.
-    event.stopPropagation();
-
-    if (!this.disabled) {
+    if (this.disabled) {
+      event.stopPropagation();
+    } else {
       this.toggle();
```
Comment (jelbourn) [inline src/components/checkbox/checkbox.ts]: This change shouldn't be part of this PR.

File: src/components/checkbox/checkbox.spec.ts [hunk 2]
```diff
@@ -403,7 +403,7 @@ describe('MdCheckbox', () => {
 
     it('should assign a unique id to each checkbox', () => {
       let [firstId, secondId] =
-          fixture.debugElement.queryAll(By.directive(MdCheckbox))
+        fixture.debugElement.queryAll(By.directive(MdCheckbox))
           .map(debugElement => debugElement.nativeElement.querySelector('input').id);
 
       expect(firstId).toBeTruthy();
```
Comment (jelbourn) [inline src/components/checkbox/checkbox.spec.ts]: You should be able to revert changes to this entire file.

File: src/core/rtl/dir.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import {Directive, HostBinding, Output, Input, EventEmitter} from '@angular/core';
 
+import {Observable} from 'rxjs/Rx';
```
Comment (DevVersion) [inline src/core/rtl/dir.ts]: Only import the Observable Library
```ts
import {Observable} from 'rxjs/Observable';
```

File: src/components/radio/radio.ts [hunk 4]
```diff
@@ -18,6 +18,7 @@ import {
   NG_VALUE_ACCESSOR,
   ControlValueAccessor
 } from '@angular/common';
+import { Observable } from 'rxjs/Rx';
```
Comment (DevVersion) [inline src/components/radio/radio.ts]: Only import the Observable Library
```ts
import {Observable} from 'rxjs/Observable';
```

File: src/components/sidenav/sidenav.ts [hunk 5]
```diff
@@ -12,6 +12,7 @@ import {
     EventEmitter,
     Renderer
 } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
```
Comment (DevVersion) [inline src/components/sidenav/sidenav.ts]: I've posted in the old PR as well, but it's still not addressed. 

You should be able to only import the `Observable` here.
```ts
import {Observable} from 'rxjs/Observable';
```

File: src/components/checkbox/checkbox.spec.ts [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import {
-    it,
```
Comment (hansl) [inline src/components/checkbox/checkbox.spec.ts]: Multi-line statements indentation is 4, not 2.

**Comment by jelbourn** [comment]
Closing this since it would be a rebase nightmare.
